Fix some spelling in the README and add some info.

blaisep opened the pull request:
@elchupanebrej , I made some small changes to the README to help with grammar and also to check my workflow.

blaisep committed:
Fix some spelling in the README and add some info.

blaisep committed:
Add stub of Leslie's tutorial with sample code files

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

elchupanebrej commented on docs/tutorial/tests/features/books.feature:
Better to put this into
tutorial>features>books.feature

And link to this file from the tests directory by `books.desktop` file

elchupanebrej replied:
Please review https://github.com/blaisep/pytest-bdd-ng/pull/3 and share your thoughts

elchupanebrej commented on docs/tutorial/tests/conftest.py:
```diff
@@ -0,0 +1,5 @@
+from steps.library_steps import *
```
`*` syntax is a bad pattern; better to list all imported steps.

elchupanebrej commented on docs/tutorial/tests/steps/library_steps.py:
```diff
@@ -0,0 +1,34 @@
+
+@when(parsers.re("a (?P<search_type>name|title) search is performed for " +
+"(?P<search_term>.+)"))
+def a_SEARCH_TYPE_is_performed_for_SEARCH_TERM(search_type, search_term):
```
`context` fixture is missing

elchupanebrej replied:
`def a_SEARCH_TYPE_is_performed_for_SEARCH_TERM(search_type, search_term, context):`

elchupanebrej committed:
Adjust tutorial to be executed during e2e test launch

elchupanebrej committed:
Slightly simplify examples

elchupanebrej commented on docs/tutorial/index.rst:
```diff
@@ -0,0 +1,19 @@
+Tutorial for Behave/Cucumber users
+==================================
+
+.. _tutorial: https://thebddcoach.com/post/a-quick-introduction-to-pytest-bdd-ng-for-people-who-are-already-familiar-with-cucumber-or-behave/
```
A better idea is to rewrite the tutorial inside a project and make it updatable by the repository owner. If we have a tutorial as part of the e2e suite - it won't become outdated.

elchupanebrej committed:
Slightly simplify examples

blaisep committed:
Merge pull request #3 from elchupanebrej/doc-e2e

Adjust tutorial to be executed during e2e test launch

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

blaisep committed:
WIP: add some diagrams to show sequence of events.

blaisep committed:
WIP: switch to sketchy theme to emphasize immaturity

blaisep committed:
WIP: tidy up the description of the examples.

The idea I had here was to describe the intent of each example and perhaps arrange them in order of complexity.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

blaisep committed:
Add a developer guide

In this commit, the guide explains how to set up the dagger client and provides a simple exercise to get you started.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
